File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        ans = nums[0]
        sum_ = nums[0]
        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                sum_ += nums[i]
                ans = max(ans, sum_)
            else:
                sum_ = nums[i]
        return ans

File: LeetCode/1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        int ans = nums[0];
        int n = nums.size();
        int sum = nums[0];
        for (int i = 1; i < n; i++) {
            if (nums[i] > nums[i-1]) {
                sum += nums[i];
                ans = max(ans, sum);
            } else {
                sum = nums[i];
            }
        }
        return ans;
    }
};